airbyte-webapp-e2e-tests/cypress/integration/connection/syncModes.spec.ts

```diff
@@ -0,0 +1,379 @@
+      saveConnectionAndAssertStreams({
+        namespace: "public",
+        name: "users",
+        config: {
+          syncMode: "full_refresh",
+          destinationSyncMode: "append",
+        },
+      });
+
+      streamsTable.checkNoSourceDefinedCursor("public", "users");
+      streamsTable.checkNoSourceDefinedPrimaryKeys("public", "users");
+    });
+  });
+
+  describe("Incremental | Deduped + history", () => {
+    describe("with source-defined primary keys", () => {
+      const cursor = "updated_at";
+      const primaryKey = "id";
+
+      before(() => {
+        streamsTable.searchStream("users2");
+        streamsTable.selectSyncMode("Incremental", "Deduped + history");
+      });
+
+      it("should be able to select cursor", () => {
+        streamsTable.hasEmptyCursorSelect("public", "users2");
+        streamsTable.selectCursor("users2", cursor);
+        streamsTable.checkSelectedCursorField("users2", cursor);
+      });
+
+      it("has source-defined primary key", () => {
+        streamsTable.checkSourceDefinedPrimaryKeys("users2", primaryKey);
+      });
+
+      it("saves", () => {
+        saveConnectionAndAssertStreams({
+          namespace: "public",
+          name: "users2",
+          config: {
+            syncMode: "incremental",
+            destinationSyncMode: "append_dedup",
+            cursorField: [cursor],
+            primaryKey: [[primaryKey]],
+          },
+        });
+
+        streamsTable.checkSelectedCursorField("users2", cursor);
+        streamsTable.checkSourceDefinedPrimaryKeys("users2", primaryKey);
+      });
+    });
+
+    describe("with source-defined cursor and primary keys", () => {
+      const cursor = "updated_at";
+      const primaryKey = "id";
+
+      before(() => {
+        streamsTable.searchStream("accounts");
+        streamsTable.selectSyncMode("Incremental", "Deduped + history");
+      });
+
+      it("has source-defined cursor", () => {
+        streamsTable.checkSourceDefinedCursor("accounts", cursor);
+      });
+
+      it("has source-defined primary key", () => {
+        streamsTable.checkSourceDefinedPrimaryKeys("accounts", primaryKey);
+      });
+
+      it("saves", () => {
+        saveConnectionAndAssertStreams({
+          namespace: "public",
+          name: "accounts",
+          config: {
+            syncMode: "incremental",
+            destinationSyncMode: "append_dedup",
+            cursorField: ["updated_at"],
+            primaryKey: [["id"]],
+          },
+        });
+
+        streamsTable.checkSourceDefinedCursor("accounts", cursor);
+        streamsTable.checkSourceDefinedPrimaryKeys("accounts", primaryKey);
+      });
+    });
+
+    describe("with selectable primary keys", () => {
+      const cursorValue = "created_at";
+      const primaryKeyValue = ["car_id", "user_id"];
+
+      before(() => {
+        streamsTable.searchStream("user_cars");
+        streamsTable.selectSyncMode("Incremental", "Deduped + history");
```

Comment:
💡 Just an idea: would be great to access sync modes via key. We can convert it to a map with tuples, for example:
const syncModes = {
      INCR_DEDUP = ["Incremental", "Deduped + history"]
      .....
}
```suggestion
        streamsTable.selectSyncMode(syncModes.INCR_DEDUP);
```

Reply:
Good idea. I will push a change